pkg/engine/internal/auth.go

```diff
@@ -0,0 +1,58 @@
+package internal
+
+import (
+	"context"
+
+	engineapi "github.com/kyverno/kyverno/pkg/engine/api"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+// Authorizer implements authorizer.Authorizer interface. It is intended to be used in validate.cel subrules.
+type Authorizer struct {
+	Client       engineapi.Client
+	ResourceKind schema.GroupVersionKind
```

Comment:
Do we need to expose these fields?

Reply:
Yes. I used to create a variable for it. 
https://github.com/kyverno/kyverno/pull/8024/files#diff-75b62813ed02d3ef4ed29adbfdd3c1c1ff4f50735af689a58e758ab54e72f0eaR118-R122

Reply:
Can you create a helper function to build it and make these fields private?

Reply:
Done.